### PR TITLE
Delete expired files every hour.

### PR DESCRIPTION
**Pull request by sudoer-zero**

### CODE REVIEW

**Comment by TobySuch** [comment]
Thanks for working on this @sudoer-zero - a few things before I merge this in:
1. The production build is failing due to pip violations. Could you fix these please:
```
Step 11/26 : RUN flake8 . --exclude=migrations
 ---> Running in 0fceaa475ad7
./shifter/settings.py:168:2: W292 no newline at end of file
./shifter_files/models.py:3:1: F401 'django.db.models.functions.Now' imported but unused
```
2. I've added in some unit tests now. Could you please add one or two for this. Should basically be the opposite of the upload_file test if you want to use that as a base.
3. I have never used this library before, but it looks like we will still need to run the following command inside the entrypoint file:
```
python manage.py crontab add
```

File: shifter/shifter_files/cron.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from django.db.models.functions import Now
+from .models import FileUpload
+
+
+def delete_expired_files():
+    FileUpload._base_manager.filter(expiry_datetime__lte=Now()).delete()
```
Comment (TobySuch) [inline shifter/shifter_files/cron.py:6]: Is there a reason this uses `_base_manager` rather than just `objects`?

**Review by TobySuch — CHANGES_REQUESTED** [review]
Thanks for working on this @sudoer-zero - a few things before I merge this in:
1. The production build is failing due to pip violations. Could you fix these please:
```
Step 11/26 : RUN flake8 . --exclude=migrations
 ---> Running in 0fceaa475ad7
./shifter/settings.py:168:2: W292 no newline at end of file
./shifter_files/models.py:3:1: F401 'django.db.models.functions.Now' imported but unused
```
2. I've added in some unit tests now. Could you please add one or two for this. Should basically be the opposite of the upload_file test if you want to use that as a base.
3. I have never used this library before, but it looks like we will still need to run the following command inside the entrypoint file:
```
python manage.py crontab add
```

File: shifter/shifter_files/cron.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from django.db.models.functions import Now
+from .models import FileUpload
+
+
+def delete_expired_files():
+    FileUpload.objects.filter(expiry_datetime__lte=Now()).delete()
+    print("deleting")
```
Comment (TobySuch) [inline shifter/shifter_files/cron.py:7]: Can you remove the debug print message please

**Review by TobySuch — CHANGES_REQUESTED** [review]
Thanks for the changes, still missing the test though.

Take a look at [this test](https://github.com/TobySuch/Shifter/blob/main/shifter/shifter_files/tests/test_views.py#L35) - your test needs to add a file to the database with an expiry datetime in the past (can just add the file to the database directly, no need to go through the post request) and then confirm that after the function in the cron.py file is ran, the file is deleted.

Let me know if you need some help with this.

File: shifter/shifter_files/cron.py (hunk 2 above)
Comment (sudoer-zero) [reply shifter/shifter_files/cron.py:7]: I didn't finish yet...
And I'm keeping tests in mind. 
I just the changes have been added automatically. SORRY!
working on it.

Comment (TobySuch) [reply shifter/shifter_files/cron.py:7]: Ah my apologies! Shoot me a message once it's done!

**Comment by sudoer-zero** [comment]
Review This.
Tell me if any changes are required!

**Review by TobySuch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**